### PR TITLE
rgw_sal_motr: [EOS-27516] Handling object overwrite during put-object

### PR DESCRIPTION
**Pull request by sjain09**
# Change Summery
## Parent Story
[CORTX-27516](https://jts.seagate.com/browse/CORTX-27516):
_PutObject API in RGW+Motr_

### Sub-tasks covered 
[CORTX-28765](https://jts.seagate.com/browse/CORTX-28765) - _Handle object overwrite_
**Changes:** Initialing existing object before creating new object. Once new objects is written successfully, deleting previous object's data in motr.

[CORTX-28776](https://jts.seagate.com/browse/CORTX-28776) - _Verify Parallel uploads_
**Changes:** Found object leak during parallel upload which we will be properly design and fixed in Jira : [CORTX-28825](https://jts.seagate.com/browse/CORTX-28825). For now, added a TODO comment in the code.

## Testing done:
<details><summary>Overwrite object</summary>

#### Command line:

> [root@ssc-vm-g4-rhev4-0499 ~]# md5sum testfile
> 8e5daee75e9a4bb3d912168d5c14f960  testfile
> [root@ssc-vm-g4-rhev4-0499 ~]# aws s3api put-object --bucket testbuck --key testfile --body testfile
> {
>     "ETag": "\"8e5daee75e9a4bb3d912168d5c14f960\""
> }
> [root@ssc-vm-g4-rhev4-0499 ~]# md5sum testfile1
> 99cf4f4bd357c9bee5daba4a4d28f5ac  testfile1
> [root@ssc-vm-g4-rhev4-0499 ~]# aws s3api put-object --bucket testbuck --key testfile --body testfile1
> {
>     "ETag": "\"99cf4f4bd357c9bee5daba4a4d28f5ac\""
> }

</details>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_sal_motr.cc [hunk 1]
```diff
@@ -2054,6 +2071,7 @@ void MotrAtomicWriter::cleanup()
   m0_bufvec_free2(&buf);
   acc_data.clear();
   obj.close_mobj();
+  old_obj.close_mobj();
```
Comment (sachinpunadikar) [inline src/rgw/rgw_sal_motr.cc:2074]: Is there any need to close the old_obj ? I suppose it was not opened at all.

Comment (sjain09) [reply src/rgw/rgw_sal_motr.cc:2074]: Right, mobj is not opened before we are actually trying to delete it. I have added for the sake of cleanup.

**Review by sachinpunadikar — APPROVED** [review]
+1

**Review by kupranay — APPROVED** [review]

**Comment by andriytk** [comment]
Check https://github.com/Seagate/cortx-rgw/issues/7 - this is how it should have been fixed.

**Comment by sjain09** [comment]
> Check #7 - this is how it should have been fixed.

We will implement this logic when we start using GC.

**Pull request merged**